File: Session7/Java/GroupAnagrams.java
https://leetcode.com/problems/group-anagrams/

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        Map<String, List<String>> map = new HashMap<>();
        
        for (String str:strs) 
        {
            char[] arr = str.toCharArray();
            Arrays.sort(arr);
            String keyStr = String.valueOf(arr);
            if(map.containsKey(keyStr))
            {
                map.get(keyStr).add(str); 
            }
            else
            {
                map.put(keyStr,new ArrayList<>());
                map.get(keyStr).add(str);
            }
        }
        
        return new ArrayList<>(map.values());
        
    }
}

File: Session7/C++/GroupAnagrams.cpp
https://leetcode.com/problems/group-anagrams/

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        unordered_map<string, vector<string>> m;
        string original;
        
        for(auto str:strs)
        {
            original = str;
            sort(str.begin(), str.end());
            m[str].push_back(original);
        }
        
        vector<vector<string>> result;
        for(auto i:m)
        {
            result.push_back(i.second);
        }
        
        return result;
        
    }
};